src/creatures/players/account/account.cpp

```diff
@@ -353,6 +354,41 @@ namespace account {
 		return ERROR_NO;
 	}
 
+	void Account::UpdatePremium() {
+		uint32_t remainingDays = 0;
+		time_t lastDay;
+		std::string accountIdentifier;
+		GetPremiumRemainingDays(&remainingDays);
+		GetPremiumLastDay(&lastDay);
+		GetAccountIdentifier(&accountIdentifier);
+		time_t currentTime = getTimeNow();
+
+		if (lastDay < currentTime) {
+			if (SetPremiumRemainingDays(0) != ERROR_NO || SetPremiumLastDay(0) != ERROR_NO) {
+				g_logger().error("Failed to reset premium from account {}: {}", getProtocolCompat() ? "name" : "email", accountIdentifier);
+			}
+		} else if (lastDay == 0) {
+			SetPremiumRemainingDays(0);
+		} else {
+			uint32_t daysLeft = static_cast<int>((lastDay - currentTime) / 86400);
+			uint32_t timeLeft = static_cast<int>((lastDay - currentTime) % 86400);
+			if (daysLeft > 0) {
+				SetPremiumRemainingDays(daysLeft);
+			} else if (daysLeft == 0 && timeLeft > 0) {
```

Comment:
In the previous if check if daysLeft is > 0, so here it will always be 0

```suggestion
			} else if (timeLeft > 0) {
```

Reply:
> 



> In the previous if check if daysLeft is > 0, so here it will always be 0

At this moment I force the check to daysLeft to be 0 because I don't know if daysLeft can be < 0.

Reply:
> > 
> 
> > In the previous if check if daysLeft is > 0, so here it will always be 0
> 
> At this moment I force the check to daysLeft to be 0 because I don't know if daysLeft can be < 0.

It cannot, it is unsigned, which means that its minimum value is 0.

Reply:
> > > 
> > 
> > 
> > > In the previous if check if daysLeft is > 0, so here it will always be 0
> > 
> > 
> > At this moment I force the check to daysLeft to be 0 because I don't know if daysLeft can be < 0.
> 
> It cannot, it is unsigned, which means that its minimum value is 0.

yeah.. it makes sense! :)